dashboard/app/controllers/aichat_controller.rb

```diff
@@ -212,4 +239,26 @@ def find_toxicity
     limit = DCDO.get('aichat_request_limit_per_min', DEFAULT_REQUEST_LIMIT_PER_MIN)
     Cdo::Throttle.throttle(AICHAT_PREFIX + id.to_s, limit, 60)
   end
+
+  private def chat_history_permission_error(student_user_id, level_id, script_id, script_level_id = nil)
```

Comment:
Nice helper function!

Reply:
+1! nit: potentially rename to something like `can_view_student_chat_history?` and return boolean?